aws/resource_aws_lb_listener_rule_test.go

```diff
@@ -2075,3 +2458,974 @@ resource "aws_security_group" "test" {
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "TestAccAWSALB_conditionPathPattern"
+  }
+}`, lbName)
+}
+
+func testAccAWSLBListenerRuleConfig_conditionPathPattern_deprecated(lbName string) string {
+	return fmt.Sprintf(`resource "aws_lb_listener_rule" "static" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+  priority = 100
+
+  action {
+    type = "fixed-response"
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "Static"
+      status_code = 200
+    }
+  }
+
+  condition {
+    field = "path-pattern"
+    values = ["/public/*"]
+  }
+}
+
+resource "aws_lb_listener" "front_end" {
+   load_balancer_arn = "${aws_lb.alb_test.id}"
+   protocol = "HTTP"
+   port = "80"
+
+   default_action {
+     type = "fixed-response"
+     fixed_response {
+       content_type = "text/plain"
+       message_body = "Not Found"
+       status_code = 404
+     }
+   }
+}
+
+resource "aws_lb" "alb_test" {
+  name            = "%s"
+  internal        = true
+  security_groups = ["${aws_security_group.alb_test.id}"]
+  subnets         = ["${aws_subnet.alb_test.*.id[0]}", "${aws_subnet.alb_test.*.id[1]}"]
+
+  idle_timeout = 30
+  enable_deletion_protection = false
+
+  tags = {
+    Name = "TestAccAWSALB_conditionPathPattern"
+  }
+}
+
+variable "subnets" {
+  default = ["10.0.1.0/24", "10.0.2.0/24"]
+  type    = "list"
+}
+
+data "aws_availability_zones" "available" {}
+
+resource "aws_vpc" "alb_test" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name = "TestAccAWSALB_conditionPathPattern"
+  }
+}
+
+resource "aws_subnet" "alb_test" {
+  count                   = 2
+  vpc_id                  = "${aws_vpc.alb_test.id}"
+  cidr_block              = "${element(var.subnets, count.index)}"
+  map_public_ip_on_launch = true
+  availability_zone       = "${element(data.aws_availability_zones.available.names, count.index)}"
+
+  tags = {
+    Name = "TestAccAWSALB_conditionPathPattern-${count.index}"
+  }
+}
+
+resource "aws_security_group" "alb_test" {
+  name        = "allow_all_alb_test"
+  description = "Used for ALB Testing"
+  vpc_id      = "${aws_vpc.alb_test.id}"
+
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "TestAccAWSALB_conditionPathPattern"
+  }
+}`, lbName)
+}
+
+func testAccAWSLBListenerRuleConfig_conditionQueryString(lbName string) string {
+	return fmt.Sprintf(`resource "aws_lb_listener_rule" "static" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+  priority = 100
+
+  action {
+    type = "fixed-response"
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "Static"
+      status_code = 200
+    }
+  }
+
+  condition {
+    field = "query-string"
+    query_string {
```

Comment:
Can we update this test to also add another block, and ensure we can have 2 side-by-side `query_string` blocks?

Reply:
Have expanded the test case. Not sure if this quite what you were after.